daemon/cluster/executor/container/controller.go

```diff
@@ -98,6 +98,13 @@ func (r *controller) Prepare(ctx context.Context) error {
 		return err
 	}
 
+	// Before we create networks, we need to make sure that the node has all of
+	// the network attachments that it needs. This will block until that is the
+	// case or the context has expired.
+	if err := r.adapter.nodeAttachmentsReady(ctx); err != nil {
+		return err
+	}
+
```

Comment:
So, please pardon my ignorance.  Three questions: 
  1. Is the Prepare() function invoked in its own goroutine and so blocking here won't unduly hold up any other operations?
  2. Alternately, will blocking this goroutine hold up some client (e.g. the swarm manager) for an excessive amount of time that might cause problems?  (because it is waiting for the task to be prepared)
  3. Is the context's timeout sufficiently long to allow node network attachments to propagate to the engine?